rotkehlchen/chain/evm/decoding/decoder.py

```diff
@@ -654,13 +685,13 @@ def _maybe_decode_erc20_721_transfer(
             transaction=transaction,
         )
         if transfer is None:
-            return None, []
+            return None, [], None
 
         for idx, action_item in enumerate(action_items):
             if action_item.asset == found_token and action_item.amount == transfer.balance.amount and action_item.from_event_type == transfer.event_type and action_item.from_event_subtype == transfer.event_subtype:  # noqa: E501
                 if action_item.action == 'skip':
                     action_items.pop(idx)
-                    return None, []
+                    return None, [], None
```

Comment:
General comment. This is why I did not want you to do this change like this. You said it's not many places when I told you that you would need to add counterparty everywhere. Yet .... here you are. Doing exactly what I todl you I wanted to avoid.

Can't scale decoders like this by having to add a new argument or return argument to all functions involved in the call chain. The code becomes unmaintenable.

The solution here is to have a context class that is passed as argument and as return. Something like `DecodersArgContext` and `DecodersReturnContext`.

You simply change the context class in one place and don't need to re-touch every single call function in the decoder call stack ever again.